---
title: "Integration of single-cell tools into GenAP2 galaxy"
output: html_document
---

```{bash setup, include=FALSE,  eval = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Purpose 

This vignette contains all the steps necessary to implement the single-cell Galaxy tools included in the GenAP2 release.

# Create necessary Galaxy environments from the command line

Depending on whether this is necessary or possible in the master version, add the required channels.

```{bash,eval = FALSE, results = FALSE}
## Add bioconda and conda-forge channels for salmon
conda config --append channels bioconda
conda config --append channels conda-forge


## r-dplyr
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create -c r --name __r-dplyr@0.7.6 r-dplyr==0.7.6

## r-feather
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create -c r --name __r-feather@0.3.1 r-feather==0.3.1

## Jemalloc
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create -c conda-forge --name __jemalloc@5.2.0 jemalloc==5.2.0

## r-base
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name __r-base@3.5.1 r-base==3.5.1


## r-matrix
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name __r-matrix@1.2_14 r-matrix==1.2_14

## r-seurat
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name __r-seurat@2.3.4 r-seurat==2.3.4

## r-optparse
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name __r-optparse@1.3.2 r-optparse==1.3.2

## r-getopt
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name __r-getopt@1.20.2 r-getopt==1.20.2

## r-workflowscriptscommon
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name __r-workflowscriptscommon@0.0.4 r-workflowscriptscommon==0.0.4

## Salmon
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create -c bioconda --name __salmon@0.13.1 salmon==0.13.1

##scanpy-scripts
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create -c bioconda --name __scanpy-scripts@0.0.5 scanpy-scripts==0.0.5 

## Install anndata in mulled environment for convert alevin to scanpy
conda activate mulled-v1-ba937bddbb3d52e12e613eb000df27216427e0bef54b600eb7648cd9ef3728db
conda install anndata --channel bioconda

/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create -c bioconda --name __salmon@0.13.1 salmon==0.13.1


## Custom mulled environment for Run-Tsne from seurat-scripts
python mulled_name.py __r-base@3.4.1 __r-seurat@2.3.4 __seurat-scripts@0.0.5 __r-optparse@1.3.2 __r-workflowscriptscommon@0.0.4

/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name mulled-v1-0f6f029f5038b7f518b01fb428b8ad1f11e9bbc03573e423361dfd1465661468 r-seurat=2.3.4 --channel bioconda
conda activate mulled-v1-0f6f029f5038b7f518b01fb428b8ad1f11e9bbc03573e423361dfd1465661468
conda install r-base=3.4.1
conda install r-optparse=1.3.2
conda install seurat-scripts=0.0.5 --channel bioconda
conda install r-workflowscriptscommon=0.0.4 --channel bioconda
conda install r-seurat=2.3.4 --channel bioconda

/cvmfs/soft.galaxy/v2.1/server/.venv/bin/python /cvmfs/soft.galaxy/v2.1/server/.venv/bin/supervisorctl -c /etc/supervisor/conf.d/galaxy.conf restart main:

## Custom mulled environment for prep_SC_for_shiny
python mulled_name.py __r-base@3.5.1 __r-getopt@1.20.3  __r-seurat@3.0.1 __r-feather@0.3.1 __r-data.table@1.10.4 __r-dplyr@0.7.6 

/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/python /cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/bin/conda create --name mulled-v1-083b11b8c4ec2fb76fdc30a0e690354da280a476c901196cf244a24db0dbb1c8 r-seurat==3.0.1 --channel bioconda
conda activate mulled-v1-083b11b8c4ec2fb76fdc30a0e690354da280a476c901196cf244a24db0dbb1c8
conda install r-feather==0.3.1 --channel conda-forge
conda install r-data.table==1.10.4 --channel conda-forge
conda install r-dplyr==0.7.6 --channel conda-forge
conda install r-getopt==1.20.3
conda install r-base==3.5.1
conda install anndata --channel bioconda
## Needs libiconv, otherwise it will fail!
conda install libiconv --channel conda-forge

conda install r-seurat==3.0.1 --channel bioconda
```


# Create references for salmon 

This will download the necessary files using ftp and then use salmon to construct these files. The resulting reference files should be put in a directory that is accesible to Galaxy and needs to be specified in the salmon galaxy tools for it to work.

```{bash,eval = FALSE, results = FALSE}

## Download references
## Human
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.transcripts.fa.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.pc_transcripts.fa.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_29/gencode.v29.primary_assembly.annotation.gtf.gz
## Mouse
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M20/gencode.vM20.transcripts.fa.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M20/gencode.vM20.pc_transcripts.fa.gz
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M20/gencode.vM20.primary_assembly.annotation.gtf.gz

## Create salmon indices
## Human
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/envs/__salmon@0.13.1/bin/salmon index -i v29_index -k 31 --gencode -p 4 -t gencode.v29.transcripts.fa.gz
## Mouse
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/envs/__salmon@0.13.1/bin/salmon index -i vM20_index -k 31 --gencode -p 4 -t gencode.vM20.transcripts.fa.gz

## Use bioawk to create a transcript to gene translation file
## Human
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/envs/__bioawk@1.0/bin/bioawk -c gff '$feature=="transcript" {print $group}' gencode.v29.primary_assembly.annotation.gtf | awk -F ' ' '{print substr($4,2,length($4)-3) "\t" substr($8,2,length($8)-3)}' -  > v29_txp2gene.tsv
## Mouse
/cvmfs/soft.galaxy/v2.1/tool-dependency/_conda/envs/__bioawk@1.0/bin/bioawk -c gff '$feature=="transcript" {print $group}' gencode.vM20.primary_assembly.annotation.gtf | awk -F ' ' '{print substr($4,2,length($4)-3) "\t" substr($8,2,length($8)-3)}' -  > vM20_txp2gene.tsv
```



#	Clone tools into galaxy and add them to the tool conf

```{bash,eval = FALSE, results = FALSE}
 ## Salmon
Git clone https://github.com/FloWuenne/salmon_alevin.git

## EBI scRNA-seq tools
Git clone https://github.com/FloWuenne/genap2_ebi_scRNAseq_tools.git
```

# Add tools to galaxies conf.xml

```{bash,eval = FALSE, results = FALSE}
## Add the tool to the tool_conf.xml
cd /cvmfs/soft.galaxy/v2.1/server/config

## Go into tool_conf.xml and add salmon_alevin in the desired category. Suggestion: 

## Salmon
  <section_name="scRNA-seq: Salmon Alevin" id="salmon_alevin" >
    <tool file ="salmon_alevin/salmon_alevin.xml"/>
    <tool file ="salmon_alevin/alevin_to_seurat.xml"/>
    <tool file ="salmon_alevin/alevin_to_scanpy.xml"/>
    <tool file ="prep_SC_for_shiny/prep_SC_for_shiny.xml"/>
  </section>

## EBI
## Add tool location to tool_conf.xml
## add the following lines
  <section name="scRNA-seq: Seurat" id="seurat-tools">
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_read10x.xml"  labels="I/O"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_create_object.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_filter_cells.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_normalise_data.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_find_variable_genes.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_scale_data.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_run_pca.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_dim_plot.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_find_clusters.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_run_tsne.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_find_markers.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/seurat/seurat_export_cellbrowser.xml"/>
  </section>
  <section name="scRNA-seq: Scanpy" id="scanpy-tools">
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-read-10x.xml" labels="I/O"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-filter-cells.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-filter-genes.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-normalise-data.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-find-variable-genes.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-scale-data.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-run-pca.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-neighbours.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-find-cluster.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-run-umap.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-run-tsne.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-find-markers.xml"/>
    <tool file="genap2_ebi_scRNAseq_tools/scanpy/scanpy-parameter-iterator.xml"/>
  </section>
  <section name="UCSC Cell Browser" id="cell-browser">
    <tool file="genap2_ebi_scRNAseq_tools/ucsc-cell-browser/cell-browser.xml"  labels="Viz"/>
  </section>
```

