---
title: "GenAp2 single-cell tutorials overview"
author: "Florian Wuennemann"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

Welcome to the tutorial pages for the single-cell RNA-seq data processing pipelines implemented in GenAP2. We aim to support a broad range of different users and therefore implemented a number of different starting points for using our pipelines.

If you are interested in the setup of the entire processing pipeline, please consult the following tutorial:

[Pipeline description](./tutorials/pipeline_description.md)

If you want to process your scRNA-seq data, choose your starting point and follow the instructions to analyze your data:

[I have raw FASTQ files for my sample](#part1)

[I have a DGE file (matrix of cells x genes) containing UMI counts](#part2)

[I have a processed Seurat object (Seurat v2.3.4)](#part3)

# Part 1: Processing FASTQ files to DGE {#part3}



# Part 2: Clustering DGE files {#part2}

# Part 3: Visualizing and analyzing clustering results {#part3}


# Acknowledgements

 ![](../www/canarie-logo.png) 
 
This project was funded by [Canarie Inc. (Ottawa, Canada)](https://www.canarie.ca/) under the grant: Research Software Program Funding Call # 2b â€“ RS-311.

 ![](../www/genap_logo.png) 

The pipelines are all integrated into the existing GenAP2 framework, which has been established by the GenAP2 team.
